### ISSUE TITLE
How to calculate the grad of input "dloss/dx"

### ISSUE THREAD

**Issue by Rlee719** [Jittor/jittor#340]
Is it possible to calculate the grad of input (dloss/dx) that passed into Module.execute()? 

I tried to initialize a placement variable in Module.__init__() and pass the input variable to it like: 

`
class My_Network(nn.Module): 
    def __init__(): 
        super(My_Network, self.).__init__()
        self.input_var = jt.zeros(batch_size, c, h, w)
        self.input_var.requires_grad = True

    def execute(self, input): 
        self.input_var = input
`
Then I query the grad during training: 

`
self.optimizer.step(loss)
self.optimizer._build_grad_map()
for pg in self.optimizer.param_groups: 
    for p, g in zip(pg["params"], pg["grads"]):      
        if p.name() == "input_var": 
            print(p.opt_grad(self.optimizer))
            print(g)
`

However the output is just zeros like jt.Var([0.], , dtype=float32)

Is there any way to calculate the grad of input correctly? Thanks in advance!
